spec/prometheus/client/counter_spec.rb

```diff
@@ -122,4 +117,57 @@
       end
     end
   end
+
+  describe '#with_labels' do
+    let(:expected_labels) { [:foo] }
+
+    it 'pre-sets labels for observations' do
+      expect { counter.increment }
+        .to raise_error(Prometheus::Client::LabelSetValidator::InvalidLabelSetError)
+      expect { counter.with_labels(foo: 'label').increment }.not_to raise_error
+    end
+
+    it 'registers `with_labels` observations in the original metric store' do
+      counter.increment(labels: { foo: 'value1'})
+      counter_with_labels = counter.with_labels({ foo: 'value2'})
+      counter_with_labels.increment(by: 2)
+
+      expect(counter_with_labels.values).to eql({foo: 'value1'} => 1.0, {foo: 'value2'} => 2.0)
```

Comment:
Could compare the two `values` with each other rather than looking for specific values in the hash. You're already testing that the thing returned by `with_labels` works above.

Reply:
you mean expect counter_with_labels.values? == counter.values?

Reply:
If you do mean that, i prefer this way, because otherwise those hashes could be empty for all i know...

Reply:
Yeah, that is what I mean.

There are other tests that assert that labels behave in a reasonable way in general. I wouldn't expect them to suddenly do something wierd (i.e. be completely wiped out in both the original and new counters) by a call to `with_labels`.

As a testing style thing, I try not to have too many tests checking the same property of the code over and over again.

Reply:
How strongly do you feel about this?
Because I think the test, as is, is more dependable (as it's more obvious what the exact contents of the hash is)
But if that's going to block merging this fix, i'll just change it.

Reply:
Really not that strong. Definitely not strongly enough to stop this fix going out. Resolving.

Reply:
Shall we merge it in then?